### PR TITLE
Move entries

### PR DESCRIPTION
**Pull request by al-serebrov**
Hey!
I've added functionality for moving entries up and down.
The things I don't like the most:
~1. (going to change this) To uncover the order of entries on the webpage I get all database, while I need only indexes~
2. Don't really like the code, it seems to be over complicated, going to enhance it a bit.
3. The route method is GET, while I'm changing the set of entries, shouldn't it be POST?
4. Didn't add an appropriate test for the new functionality, have the same question as for delete-entry test. There's no `flaskr_tests.py` file set up in this branch, what's the workflow in this case?
5. Some bugs could be found while testing this along with the `delete_entry()` route. Hopefully none of them but definitely needs to be tested.

But in the meantime, it'd be really nice if you have a quick look at this branch.

Thank you in advance.

### CODE REVIEW

File: flaskr.py
Comment (serebrov) [inline flaskr.py:18]: The formatting change is not neccessary here as we know that it is done on another branch. In a perfect world, the formatting would be done on the separate branch, so PRs don't get cluttered with irrelevant diffs.

This is just a notice, feel free to keep this formatting change, just don't do such repeated change on multiple branches in the future.

File: flaskr.py
Comment (serebrov) [inline flaskr.py]: Unfortunately this is not appropriate. I am even shocked a bit. :scream: 
You should never change the ID of the record. This is the unique identifier you assign to it when it's created and it stays same forever (or until it gets deleted).

But even if the record is deleted, no other record should acquire the previously used ID.

Comment (serebrov) [reply flaskr.py]: The way to go is to add a separate ordering field, for example `sort_order`.

Other notes:
- There should be no magical constants such as 99999999
- There should be no special cases for move up and down, try to think on the more generic algorithm
- The algorithm should not get all the records from the DB, it should only use current record and then find the record before / after it (depending on the direction) and swap them

File: templates/show_entries.html
Comment (serebrov) [inline templates/show_entries.html]: Yes, it is better to have them as POSTs.

Two solutions here: use the form with button (you can still style it as a link) or simple jQuery script to make this link perform the POST request (the link will have no "href" and the actual work will be done in the onclick event handler).

**Review by serebrov — CHANGES_REQUESTED** [review]
Nice change, please check my comments ;)

**Comment by serebrov** [comment]
> 4. Didn't add an appropriate test for the new functionality, have the same question as for delete-entry test. There's no flaskr_tests.py file set up in this branch, what's the workflow in this case?
> 5. Some bugs could be found while testing this along with the delete_entry() route. Hopefully none of them but definitely needs to be tested.

Two ways here:
1) Wait until the file you need gets merged to master and then merge master to your branch
2) Just add this file to your branches (and only put the new code to it, don't copy the work from other branches) - it will be merged later, that's why we use git, it handles these cases. The merge will probably result in the merge conflict, but it will be trivial to solve it.

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py:18]: Yes, I know right. That's because of the original file had tab-style indentation and I converted it to spaces in order to write space-indented code.

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py]: ~I figured out almost everything that you asked for, except really generic algorithm, it seems that I'm pretty close to it, but not sure if I changed the code correctly, please check the last commit before I continue working on this.~

Nevermind this for now, I just noticed that I'm still getting all records' ids and sort_order :D need more time to work on this, I'll keep you posted, thanks.

Comment (al-serebrov) [reply flaskr.py]: This one is fixed except:
>There should be no special cases for move up and down, try to think on the more generic algorithm

File: flaskr.py
Comment (serebrov) [inline flaskr.py]: select id, sort_order from entries where sort_order > (?) order by sort_order asc limit 1;

File: flaskr.py
Comment (serebrov) [inline flaskr.py]: select id, sort_order from entries where sort_order < (?) order by sort_order desc limit 1;

File: flaskr.py
Comment (serebrov) [inline flaskr.py:181]: This is correct, but you probably also want to handle the edge cases - when there is no record with sort_order less then current or greater than current.

File: flaskr.py
Comment (serebrov) [inline flaskr.py]: Why `10`? Why not `78` or something else? I would expect `1` which is not a magic number (https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants).

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py]: Corrected

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py]: Corrected

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py:181]: Added handling through python errors handling, please let me know if it's incorrect somehow

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py]: Thanks for the info, corrected this

File: templates/show_entries.html
Comment (al-serebrov) [reply templates/show_entries.html]: Unfortunately, I'm not familiar with jQuery, tried a couple options that I had googled but unsuccessfully because I have no idea what's happening inside the code. Postponed getting familiar with jQuery for some time :D for now switched the link into the form.
Don't really like the "design", going to change this when all three branches get merged.

File: flaskr.py
Comment (serebrov) [inline flaskr.py]: This is still magic "10", why 10 and not something else?

File: flaskr.py
Comment (serebrov) [reply flaskr.py:181]: I am not sure in which case each error could happen, it deserves at least a comment.

File: flaskr.py
Comment (serebrov) [inline flaskr.py]: This is not good. The purpose of the transaction is to make the change atomic - it whether passes as a whole or fails as a whole. So we should do "commit" only after the second statement, this way we will update both records or update no records at all. Now it is potentially possible to "corrupt" the data if we change the sort order for first record and second update fails.

**Review by serebrov — APPROVED** [review]
LGTM, there are still some minor comments.

**Comment by serebrov** [comment]
@al-serebrov also there are some conflicts, you need to merge master to this branch and resolve them. And you need to remove the pyc file from the source control, and probably flaskr.db too (it is better to keep it as SQL and reconstruct when needed into db).

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py]: :flushed: 
no way, I did change it to 1, very weird

File: flaskr.py
Comment (al-serebrov) [reply flaskr.py]: Yes, I agree, I was wondering do I need two db.commit() in the code, just forgot to ask :) thanks for the clarification

**Pull request merged**
